### PR TITLE
vim-patch:8.1.1763: evalfunc.c is still too big

### PR DESCRIPTION
**Pull request by lewis6991**
Problem:    Evalfunc.c is still too big.
Solution:   Move dict and list functions to a better place.
https://github.com/vim/vim/commit/9f9fe37f6750f342255a51f158a7bb372c827f7f

### CODE REVIEW

**Comment by zeertzjq** [comment]
Did you mess up a rebase?

**Comment by lewis6991** [comment]
Yes, should be ok now

**Comment by clason** [comment]
nice try, though ;)

File: src/nvim/eval/funcs.h [hunk 1]
```diff
@@ -4,8 +4,6 @@
 #include "nvim/buffer_defs.h"
 #include "nvim/eval/typval.h"
 
-typedef void (*FunPtr)(void);
-
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.h:8]: I don't think this should be moved.

Comment (lewis6991) [reply src/nvim/eval/funcs.h:8]: It is required in `typval.h` and `funcs.h` includes that. How else could we organise this?

**Comment by zeertzjq** [comment]
Oh, this may indeed be a bit hard to decide, as eval/typval was initially intended for "base typval_T manipulations" (<https://github.com/neovim/neovim/issues/5081#issuecomment-234772243>), and doesn't include many headers.

**Comment by zeertzjq** [comment]
Maybe `FunPtr` can be moved to `nvim/types.h` instead?

**Comment by zeertzjq** [comment]
Missing two comment changes above `get_literal_key()` and `dict_get_tv()`

**Review by zeertzjq — APPROVED** [review]
LGTM. Third argument of `tv_blob_remove()` is unused, but keeping it looks more consistent, so I'm not sure about this.

**Comment by lewis6991** [comment]
Let's keep the vim-patch *pure* and worry about refactoring separately. The typval stuff is mostly untouched by us so there's probably more benefit in being consistent just so future vim patches are easier to apply.

**Comment by zeertzjq** [comment]
Well, actually it will be used in patch 8.2.3284, which hasn't been ported yet.

**Pull request merged**